code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +626,237 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
 	}
 }
 
+const char *xwi_determine_object_type(const XWMObject *oj)
+{
+	switch (oj->objectType) {
+	case XWMObjectType::oj_Mine1:
+		return "Defense Mine#Ion";
+	case XWMObjectType::oj_Mine2:
+		return "Defense Mine#Ion";
+	case XWMObjectType::oj_Mine3:
+		return "Defense Mine#Ion";
+	case XWMObjectType::oj_Mine4:
+		return "Defense Mine#Ion";
+	case XWMObjectType::oj_Satellite:
+		return "Sensor Satellite#Imp";
+	case XWMObjectType::oj_Nav_Buoy:
+		return "Nav Buoy#real";
+	case XWMObjectType::oj_Probe:
+		return "Sensor Probe";
+	case XWMObjectType::oj_Asteroid1:
+		return "Asteroid#Small01";
+	case XWMObjectType::oj_Asteroid2:
+		return "Asteroid#Small02";
+	case XWMObjectType::oj_Asteroid3:
+		return "Asteroid#Medium01";
+	case XWMObjectType::oj_Asteroid4:
+		return "Asteroid#Medium02";
+	case XWMObjectType::oj_Asteroid5:
+		return "Asteroid#Medium03";
+	case XWMObjectType::oj_Asteroid6:
+		return "Asteroid#Big01";
+	case XWMObjectType::oj_Asteroid7:
+		return "Asteroid#Big02";
+	case XWMObjectType::oj_Asteroid8:
+		return "Asteroid#Big03";
+	case XWMObjectType::oj_Training_Platform1:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform2:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform3:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform4:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform5:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform6:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform7:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform8:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform9:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform10:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform11:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform12:
+		return nullptr;
```

Comment:
Just delete all these lines.  The function will return nullptr for any enum that isn't part of the switch, and by removing these lines you can reduce clutter and focus on the significant parts.